lib/nrf91_lwm2m_app/Kconfig

```diff
@@ -10,25 +10,28 @@ config NRF91_LWM2M_APP
 
 if NRF91_LWM2M_APP
 
-config LWM2M_CONN_MON_OBJ_SUPPORT
+config NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT
 	bool "Connection Monitor object support"
 	default y
+	select LWM2M_CONN_MON_OBJ_SUPPORT
```

Comment:
Note, if a user uses menuconfig and sets `NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT=n` then LWM2M_CONN_MON_OBJ_SUPPORT will still be enabled.

Not sure that is intended ?

Also, what is the reason for having both symbols ?
In code and CMake , it seems that only NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT is used.

See also: https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig

Reply:
> Also, what is the reason for having both symbols ?

`LWM2M_CONN_MON_OBJ_SUPPORT` indicates that the LwM2M library supports the Connectivity Monitoring object (supports but does not implement the object instance). 
`NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT` indicates that this library provides a default implementation of the Connectivity Monitoring object. The user should still be able to disable the default implementation and provide his own.

Reply:
> LWM2M_CONN_MON_OBJ_SUPPORT indicates that the LwM2M library supports the Connectivity Monitoring object 

So if this is something the library indicates that it supports, then it sounds like `LWM2M_CONN_MON_OBJ_SUPPORT` itself probably should be promptless ?
Someone providing an implementation, like `NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT` can then still select that symbol.
At least from your description and my interpretation, it sounds like it should not be the user who configures `LWM2M_CONN_MON_OBJ_SUPPORT`.

Reply:
> Note, if a user uses menuconfig and sets `NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT=n` then LWM2M_CONN_MON_OBJ_SUPPORT will still be enabled.
... 
> See also: https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig

For me it looks like that issue applies only if I would do something like

```
config LWM2M_CONN_MON_OBJ_SUPPORT_FOO
	bool "Foo"

config NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT
	bool "Connection Monitor object support"
	default y if LWM2M_CONN_MON_OBJ_SUPPORT_FOO
	default n
```

But that's not the case here.

Reply:
Well, I might lack Kconfig knowledge here, but I always thought that `select` guarantees that the user does not have to bother with configuring `LWM2M_CONN_MON_OBJ_SUPPORT` on his own.

`LWM2M_CONN_MON_OBJ_SUPPORT` is an upstream symbol, it should not depend on `NRF91_LWM2M_APP` library in any way. I don't know how much flexibility do we have in reconfiguring upstream Kconfig entries from `sdk-nrf` itself.

Reply:
> LWM2M_CONN_MON_OBJ_SUPPORT is an upstream symbol, 

I didn't notice that, that of course  makes it different. I thought it was a NCS defined symbol.

>  it should not depend on NRF91_LWM2M_APP library in any way. 

Agree.

> Well, I might lack Kconfig knowledge here, but I always thought that select guarantees that the user does not have to bother with configuring LWM2M_CONN_MON_OBJ_SUPPORT on his own.

Yes, as long as `NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT=y`, but if user uses `menuconfig` to disable `NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT`, then `LWM2M_CONN_MON_OBJ_SUPPORT` will still be true.

Hence the reason for initial comment.

Reply:
> but if user uses menuconfig to disable NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT, then LWM2M_CONN_MON_OBJ_SUPPORT will still be true.

My interpretation is that the are no issues left anymore. Only menuconfig here should be the one which disables/enables the library.

Reply:
I assume there are no objections left.

Reply:
as discussed, having LWM2M_CONN_MON_OBJ_SUPPORT=y and NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT=n which happens when user deselects NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT should not be an issue in this case.